kategory/src/main/kotlin/kategory/typeclasses/MonadReader.kt

```diff
@@ -11,5 +11,11 @@ interface MonadReader<F, D> : Monad<F> {
     fun <A> reader(f: (D) -> A): HK<F, A> = map(ask(), f)
 }
 
+inline fun <reified F, A, reified D> HK<F, A>.local(FT: MonadReader<F, D> = monadReader(), noinline f: (D) -> D): HK<F, A> =
```

Comment:
Yes, that compiles though it fails at runtime because Option has no MonadReader instance which is one of the challenges we face because our instance lookups are at runtime since Kotlin lacks implicits or any proper support for typeclasses.